be/src/exec/local_file_reader.cpp

```diff
@@ -59,16 +59,22 @@ bool LocalFileReader::closed() {
 }
 
 // Read all bytes
-Status LocalFileReader::read_one_message(std::unique_ptr<uint8_t[]>* buf, size_t* length, size_t padding) {
+Status LocalFileReader::read_one_message(std::unique_ptr<uint8_t[]>* buf, size_t* capacity, size_t* length,
+                                         size_t padding) {
     bool eof;
     int64_t file_size = size() - _current_offset;
     if (file_size <= 0) {
         buf->reset();
         *length = 0;
         return Status::OK();
     }
+
+    if (*capacity < file_size + padding) {
+        buf->reset(new uint8_t[file_size + padding]);
```

Comment:
alloc one large memory as file size?

Reply:
The expected behaviour of `LocalFileReader::read_one_message` is reading all bytes.

Reply:
if the json file size is 1G, we will alloc one 1G buffer?